tests/rptest/tests/topic_creation_test.py

```diff
@@ -210,6 +210,49 @@ def test_create_topic_with_single_configuration_property(self):
             cfgs = rpk.describe_topic_configs(topic=name)
             assert str(cfgs[p][0]) == str(property_value)
 
+    @cluster(num_nodes=3)
+    @matrix(cloud_storage_enable_remote_read=[True, False],
+            cloud_storage_enable_remote_write=[True, False],
+            topic_remote_read=[True, False, None],
+            topic_remote_write=[True, False, None])
+    def test_shadow_indexing_mode(self, cloud_storage_enable_remote_read,
+                                  cloud_storage_enable_remote_write,
+                                  topic_remote_read, topic_remote_write):
+        to_bool = lambda x: True if x == "true" else False
```

Comment:
nit: this could be simplified to:
```suggestion
        to_bool = lambda x: x == "true"
```

or maybe even compare directly:

``read = `ret["redpanda.remote.read"][0] == "true"```